### ISSUE TITLE
Inconsistent virtual method obfuscation state detected

### ISSUE THREAD

**Issue by lextudio-support** [obfuscar/obfuscar#513]
This is the error:

Renaming:  fields...parameters...properties...events...methods...
An error occurred during processing:
Inconsistent virtual method obfuscation state detected. Abort. Please review the
 following methods,
[yWorks.yFilesNET.Algorithms]yWorks.yFiles.Algorithms.Geometry.IPlaneObject::get
_BoundingBox[0]->WillRename:A
[yWorks.yFilesNET.Algorithms]yWorks.yFiles.Algorithms.Geometry.LineSegment::get_
BoundingBox[0]->WillRename:A
[yWorks.yFilesNET.Algorithms]yWorks.yFiles.Algorithms.Geometry.YOrientedRectangl
e::get_BoundingBox[0]->WillRename:A
[yWorks.yFilesNET.Algorithms]yWorks.yFiles.Algorithms.Geometry.YRectangle::get_B
oundingBox[0]->WillRename:A
[yWorks.yFilesNET.Algorithms]yWorks.yFiles.Layout.Router.BusRepresentations/c::g
et_BoundingBox[0]->Skipped:special name
[yWorks.yFilesNET.Algorithms]yWorks.yFiles.Layout.LabelCandidate::get_BoundingBo
x[0]->WillRename:A
[yWorks.yFilesNET.Algorithms]or/c::get_BoundingBox[0]->WillRename:A
[yWorks.yFilesNET.Algorithms]uw/g::get_BoundingBox[0]->WillRename:A
[yWorks.yFilesNET.Algorithms]r8/a::get_BoundingBox[0]->WillRename:A
[yWorks.yFilesNET.Algorithms]ki/d::get_BoundingBox[0]->WillRename:A
[yWorks.yFilesNET.Algorithms]r8/d::get_BoundingBox[0]->WillRename:A
[yWorks.yFilesNET.Algorithms]r8/e::get_BoundingBox[0]->WillRename:A
[yWorks.yFilesNET.Algorithms]yWorks.yFiles.Layout.Router.Polyline.DynamicObstacl
eDecomposition/a::get_BoundingBox[0]->WillRename:A
[yWorks.yFilesNET.Algorithms]yWorks.yFiles.Layout.Circular.CircularLayouter/b::g
et_BoundingBox[0]->WillRename:A
[yWorks.yFilesNET.Algorithms]dg/d::get_BoundingBox[0]->WillRename:A
[yWorks.yFilesNET.Algorithms]yWorks.yFiles.Layout.Organic.RemoveOverlapsLayoutSt
age/b::get_BoundingBox[0]->WillRename:A
[yWorks.yFilesNET.Algorithms]yWorks.yFiles.Layout.Organic.ShuffleLayouter/a::get
_BoundingBox[0]->WillRename:A
[yWorks.yFilesNET.Algorithms]yWorks.yFiles.Layout.Router.Polyline.EdgeRouter/e::
get_BoundingBox[0]->WillRename:A
[yWorks.yFilesNET.Algorithms]kz/c::get_BoundingBox[0]->WillRename:A
[yWorks.yFilesNET.Algorithms]yWorks.yFiles.Layout.Labeling.MISLabelingAlgorithm/
c::get_BoundingBox[0]->WillRename:A

In the obfuscar.xml file i'm using this variables:

  <Var name="InPath" value="..\Dlls" />
  <Var name="OutPath" value="..\Dlls" />
  <Var name="KeepPublicApi" value="false" />
  <Var name="HidePrivateApi" value="true" />
  <Var name="RenameProperties" value="true" />
  <Var name="RenameEvents" value="true" />
  <Var name="ReuseNames" value="false" />

Can you help me with this issue?

**Original Reported Date**: 2014-08-18T06:10:58.117-07:00
**Planned For Release**: 
**Original CodePlex ID**: 19

**Comment by lextudio-support** [comment]
**Copied from CodePlex without authors**:

If you can share the assembly via support@lextm.com I can do some debugging on it.

Generally speaking, you can see from line 656 that only in rare cases this skip line is used,

https://github.com/lextm/obfuscar/blob/master/Obfuscar/AssemblyInfo.cs

**Original Posted Date**: 2014-08-18T20:39:51.51-07:00

**Comment by lextudio-support** [comment]
**Copied from CodePlex without authors**:

I am trying to use Obfuscar with the yWorks dlls and am getting this exact same error. I don't believe it is a duplicate of item 16.

**Original Posted Date**: 2015-02-18T10:55:46.52-08:00

**Comment by lextudio-support** [comment]
**Copied from CodePlex without authors**:

Looking at yWorks.yFilesNET.Algorithms assembly in ILSpy shows that within BusRepresentations.d (where the special name skip shows up) this property shows up as:
```
public virtual YRectangle get_BoundingBox()
{
	return this.c;
}
```

Other implementations of the IPlaneObject interface look like normal properties. Like this one from LabelCandidate:

```
public virtual YRectangle BoundingBox
{
	get
	{
		YRectangle yRectangle = LabelCandidate.a(this.a);
		if (yRectangle != null)
		{
			return yRectangle;
		}
		return this.a.BoundingBox;
	}
}
```

I've tried adding this ForceMethod element to my config, but it doesn't resolve the issue:

```
<ForceMethod type="yWorks.yFiles.Layout.Router.BusRepresentations/d" name="get_BoundingBox" />
```

I don't think the yWorks license allows me to send you their partially obfuscated dll.

**Original Posted Date**: 2015-02-18T11:03:21.247-08:00

**Comment by lextudio-support** [comment]
Duplicate to http://obfuscar.codeplex.com/workitem/16

**Original Closed Date**: 2014-09-03T06:52:28.647-07:00
